### PR TITLE
GEODE-8487: Document Redis_API HA model, requirements, guarantees

### PR DESCRIPTION
**Pull request by jhutchison**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Review by davebarnes97 — CHANGES_REQUESTED** [review]
Looks good. I'd like to request two minor changes in the file redis_api_for_geode.html.md.erb:
- In the heading "High Availability Model, Guarantees & Redis Client Responsibilities", replace the ampersand with the word "and"
- In the sentence "As persistence is not currently supported,..."  drop the word "currently". This constitutes a promise to change the functionality at an unspecified future date. Let's not go there.
Thanks for contributing to the docs!

File: geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegion.java [hunk 1]
```diff
@@ -2999,6 +2999,7 @@ private boolean putInBucket(final InternalDistributedMember targetNode, final In
 
       try {
         final boolean isLocal = (this.localMaxMemory > 0) && currentTarget.equals(getMyId());
+        logger.info(primaryMoveReadLockAcquired);
```
Comment (dschneider-pivotal) [inline geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegion.java:3002]: remove this logger statement. I think it broke your build

**Review by dschneider-pivotal — CHANGES_REQUESTED** [review]

File: geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegion.java (hunk 1 above)
Comment (jhutchison) [reply geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegion.java:3002]: thanks!

**Comment by jhutchison** [comment]
> Looks good. I'd like to request two minor changes in the file redis_api_for_geode.html.md.erb:
> 
> * In the heading "High Availability Model, Guarantees & Redis Client Responsibilities", replace the ampersand with the word "and"
> * In the sentence "As persistence is not currently supported,..."  drop the word "currently". This constitutes a promise to change the functionality at an unspecified future date. Let's not go there.
>   Thanks for contributing to the docs!

done.  thanks

**Review by dschneider-pivotal — APPROVED** [review]

**Pull request merged**